File: FrontEnd/src/app/features/boats/classes/view-models/boat-browser-list-vm.ts
import { SimpleEntity } from "src/app/shared/classes/simple-entity"
import { BoatReadFishingLicenceDto } from "../dtos/boat-read-fishing-licence-dto"
import { BoatReadInsuranceDto } from "../dtos/boat-read-insurance-dto"

export interface BoatBrowserListVM {

    beam: string
    boatUsage: SimpleEntity
    description: string
    draft: string
    fishingLicence: BoatReadFishingLicenceDto
    flag: string
    hullType: SimpleEntity
    id: number
    insurance: BoatReadInsuranceDto
    isActive: boolean
    isAthenian: boolean
    isFishingBoat: boolean
    loa: string
    postAt: string
    postUser: string
    putAt: string
    putUser: string
    registryNo: string
    registryPort: string
    
}

File: API/Features/Boats/Admin/ViewModels/BoatBrowserListVM.cs
using API.Infrastructure.Classes;

namespace API.Features.Boats.Admin {

    public class BoatBrowserListVM {

        public int Id { get; set; }
        public string Description { get; set; }
        public SimpleEntity HullType { get; set; }
        public SimpleEntity Usage { get; set; }
        public BoatBrowserListInsuranceVM Insurance { get; set; }
        public string Flag { get; set; }
        public decimal Loa { get; set; }
        public decimal Beam { get; set; }
        public decimal Draft { get; set; }
        public string RegistryPort { get; set; }
        public string RegistryNo { get; set; }
        public bool IsAthenian { get; set; }
        public bool IsFishingBoat { get; set; }
        public bool IsActive { get; set; }

    }

}